datafusion/core/src/datasource/physical_plan/mod.rs

```diff
@@ -90,6 +91,9 @@ pub struct FileSinkConfig {
     pub table_partition_cols: Vec<(String, DataType)>,
     /// Controls whether existing data should be overwritten by this sink
     pub overwrite: bool,
+    /// Optional, additional metadata to be inserted into the key_value_metadata
+    /// for the written [`FileMetaData`](https://docs.rs/parquet/latest/parquet/file/metadata/struct.FileMetaData.html).
+    pub key_value_metadata: Option<Vec<KeyValue>>,
```

Comment:
I choose to add the `key_value_metadata` to within the FileSinkConfig. An alternative could be adding another prop to the [TableParquetOptions](https://github.com/apache/datafusion/blob/06895157e7f985fc4d9b0b6298c07d92abb4cc07/datafusion/common/src/config.rs#L1368) outside of the (parsed) global.